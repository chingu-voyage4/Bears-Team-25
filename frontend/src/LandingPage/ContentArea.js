import React from 'react';


// TODO :  figure out how to grab raw text content and/or HTML of github wiki https://github.com/Chingu-cohorts/voyage-wiki/wiki

export default class ContentArea extends React.Component{
		
	constructor(props){
		super(props);
		
		this.state = {
			innerHTML: undefined
		};
		
		
		this.fetchData = this.fetchData.bind(this);
	}
	
	fetchData(selectedUrl){
		// url which is currently selected in sidebar navigation menu
		// ....
		// maybe should extract this function into its own module and update all references  
		
		let urlToGrabData = selectedUrl || 'https://cors-anywhere.herokuapp.com/https://loripsum.net/api';
		//uses lorem ipsum for example now , later will grab data from github chingu
		
		      
		let self = this;
		
		// AJAX example version: 
		fetch(urlToGrabData)
						.then(   function(response) {return response.text();}    )
						.then(   function(responseText) { self.setState( {innerHTML: responseText});     });
		
		return '<p>loading......</p>';       //line runs immediately, displays text while AJAX request in process
	}
	
	componentWillMount(){
		let grabbedText = this.fetchData();      //   fetchData(this.props.selectedUrl) ?
		this.setState({innerHTML: grabbedText});
	}	
	
	// componentWillUpdate  (){....}    -- copy/paste?    -if user clicks on another link in sidebar: need reload from new url
	
	render(){
		
		return (
			<section className='content-area' dangerouslySetInnerHTML={ {__html : this.state.innerHTML} }>
				
			</section>
		);
	}
	
	
	
}