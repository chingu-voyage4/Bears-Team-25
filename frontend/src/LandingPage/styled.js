import styled from 'styled-components';

export const Article = styled.article`
  margin: 0 auto auto;
  max-width: 60rem;
  padding: 3em 2rem;
`;

export const Heading1 = styled.h1`
  text-align: center;
`;

export const Heading2 = styled.h2`
  width: 100%;
  text-align: center;
  font-weight: 400;
  font-size: 2rem;
  color: #ff5a5f;
`;

export const Heading3 = styled.h3`

  display: flex;
  align-items: center;
  color: #7d93aa;
  font-size: 1.4em;
  font-weight: 400;
  margin-bottom: .75rem;
  margin-top: 1rem;
`;

export const Content = styled.div`
  display: flex;
  flexWrap: wrap;
  flexDirection: row;
  justifyContent: center;
`;

export const Section = styled.section`
  width: calc(50% - 1rem);
  fontSize: 1.1rem;
`;
