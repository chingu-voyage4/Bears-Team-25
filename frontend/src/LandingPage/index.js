// each container uses an index.js file to expose it's reuseable parts
import LandingPage from './LandingPage';

export default LandingPage;
