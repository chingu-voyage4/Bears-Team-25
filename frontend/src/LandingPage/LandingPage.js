// This file shows the basic layout for a react component
// It's a bit of a contrived example because airbnb linter is serously stict
// ... but that's a good thing :)
// NORMALLY, I would create a branch: `feature/LandingPage` to do this work.
// I would push the branch at the end of each day, regardless of how complete it is.
// When I think it is working and ready to be merged (into develop), I would create
// a pull request in github for the branch and add all team members as reviewers.
// BEFORE creating a pull request, run `npm run test` and `npm run lint`. If they
// both run error free, create the pull request, otherwise, fix any problems.
// SHOULD i add this to the dox mission statement workflow section?

import React from 'react';
import styled from 'styled-components';

import ContentArea from '../ContentArea/ContentArea.js';

export default class LandingPage extends React.Component {
  // define some state
  state = {
    myAlign: 'center'
  };
  // define a class member function
  testFunc = () => {
    const test = this.state.myAlign;
    return test;
  };
  render = () => {
    const Heading = styled.h1`
      text-align: ${this.testFunc()};
    `;
    return (
      <div>
        <Heading>Welcome to chingu CMS</Heading>
        <ContentArea> </ContentArea>
      </div>
    );
  };
}
