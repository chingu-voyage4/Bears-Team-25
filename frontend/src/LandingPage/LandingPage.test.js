import React from 'react';
import renderer from 'react-test-renderer';
import LandingPage from './LandingPage';
import 'jest-styled-components';

test('renders LandingPage', () => {
  const pg = renderer.create(<LandingPage />).toJSON();
  expect(pg).toMatchSnapshot();
});
