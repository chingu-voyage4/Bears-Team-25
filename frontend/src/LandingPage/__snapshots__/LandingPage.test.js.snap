// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LandingPage 1`] = `
Array [
  <h1
    className="sc-htpNat dEQUjC"
>
    Welcome to chingu CMS
</h1>,
  <article
    className="sc-bdVaJa dEZJXY"
>
    <h2
        className="sc-bxivhb iyYSDG"
    >
        Features
    </h2>
    <div
        className="sc-bwzfXH gneseq"
    >
        <section
            className="sc-EHOje lhLgo"
        >
            <h3
                className="sc-ifAKCX hJXiLo"
            >
                Manage Content
            </h3>
            <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sin aliud quid voles, postea.
            </p>
        </section>
        <section
            className="sc-EHOje lhLgo"
        >
            <h3
                className="sc-ifAKCX hJXiLo"
            >
                Search Content
            </h3>
            <p>
                Certe nihil nisi quod possit ipsum propter se iure laudari. Mihi vero, inquit, placet agi subtilius et, ut ipse dixisti, pressius.
            </p>
        </section>
        <section
            className="sc-EHOje lhLgo"
        >
            <h3
                className="sc-ifAKCX hJXiLo"
            >
                Some Feature
            </h3>
            <p>
                Omnes enim iucundum motum, quo sensus hilaretur. Duo Reges: constructio interrete. Itaque haec cum illis est dissensio, cum Peripateticis nulla sane. Quis suae urbis conservatorem Codrum, quis Erechthei filias non maxime laudat?
            </p>
        </section>
        <section
            className="sc-EHOje lhLgo"
        >
            <h3
                className="sc-ifAKCX hJXiLo"
            >
                Track Changes
            </h3>
            <p>
                Sit, inquam, tam facilis, quam vultis, comparatio voluptatis, quid de dolore dicemus? Non est igitur summum malum dolor.
            </p>
        </section>
    </div>
</article>,
]
`;
